ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,223 @@
+- **namespace** - The rego namespace of the test policy. 
+- **successes** - The number of successful checks in the form of an integer
+- **failures** - A JSON list containing objects describing each failure
+
+The **failures** list of objects will provide the following information about each check:
+- **name** - Name of the individual check written in [snake case](https://en.wikipedia.org/wiki/Snake_case)
+- **msg** - Message about what went wrong with the check
+- **description** - The explanation about why the check is significant (Why the label is used, why the vulnerabilities are checked etc.)
+- **url** - link to the further documentation on the individual check (or that type of check)
+
+Example full JSON test output for a Tekton task that tests the container image labels:
+```
+[
+    {
+        "filename": "image-inspect.json",
+        "namespace": "image_labels",
+        "successes": 19,
+        "failures": [
+            {
+                "msg": "The 'architecture' label is required",
+                "metadata": {
+                    "details": {
+                        "description": "Architecture the software in the image should target.",
+                        "name": "architecture_label_required",
+                        "url": "https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/guide_to_layered_image_build_service_osbs#jive_content_id_Labels"
+                    }
+                }
+            }
+          ]
+    }
+]
+```
+
+### Information injection
+
+Whenever possible, resources like ConfigMaps or Secrets will be used to inject configuration into Tasks. This is preferred to templates and patches as it fits well with Kubernetes declarative and GitOps approaches.
+
+ConfigMaps and Secrets will be mounted into the Task pod to inject file-based information like certificates.
+Environment variables may be injected from ConfigMaps.
+
+Variables directly configured in the Pod definition are discouraged.
+
+Side note: Secrets and ConfigMaps should be made [immutable](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable)
+
+
+### Information format
+
+[Clearly define the format of input parameters and results](https://github.com/tektoncd/catalog/blob/main/recommendations.md#clearly-define-the-format-of-input-parameters-and-results).
+
+Atomic information may be passed as a simple parameter. Non binary encoded complex information should be exchanged through [JSON format](https://github.com/tektoncd/catalog/blob/main/recommendations.md#use-composable-parameter-formats) between Tasks.
+
+#### Image references
+
+Since tags can be moved from one image to another, they should not be relied on as a reference. In order to guarantee that any scanning is performed on an image built as part of a PipelineRun, the immutable image digest reference will be used instead.
+
+#### Logging
```

Comment:
I suggest deleting this whole `#### Logging` section. I think it is duplicated by https://github.com/redhat-appstudio/book/pull/21.

Reply:
Removed. Thanks!